### PR TITLE
fix(tui): indicate clipped join tokens

### LINKED ISSUE #964: Join token is too long to read
## Problem

The join token for a private mesh is too long to see on screen, and there's no UI indication that it's been cut off.

I expected ellipsis if it's too long to fit, e.g. "abcd123..." or line wrapping.

## Steps to reproduce

1. Run the curl pipe installer
2. Run mesh-llm serve --model Qwen3-8B-Q4_K_M
3. Try to type the token on another node. The token is actually too wide for my entire 1920x1080 screen to show.

## Diagnostics

N/A

## Extra context

Image

### PR DESCRIPTION
**Pull request by Bortlesboat**
## Summary

- show an ellipsis when the join token continues beyond the visible panel width
- show indicators on both edges while horizontally scrolled
- preserve the existing left/right and `g`/`G` navigation behavior

Closes #964.

## Validation

- `cargo test -p mesh-llm-tui join_token_slice --lib`
- `cargo test -p mesh-llm-tui tui_join_token --lib`
- `cargo fmt --all -- --check`
- `git diff --check`

The repository's `just with-lld` wrapper could not run in this Windows environment because its configured Unix shell is unavailable, so the same focused Cargo commands were run directly.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved join-token display in the terminal interface by adding ellipsis indicators when content is truncated.
  * Updated slicing behavior for edge cases, including empty tokens and very narrow widths, to ensure consistent, accurate rendering.
* **Tests**
  * Added unit tests covering left/right truncation, empty-token handling, and narrow-width ellipsis behavior.

### CODE REVIEW

File: crates/mesh-llm-tui/src/output/mod.rs [hunk 1]
```diff
@@ -4728,7 +4728,32 @@ fn join_token_char_count(token: &str) -> usize {
...
+    let mut visible = String::with_capacity(width);
+    if hidden_left {
+        visible.push('…');
+    }
+    let content_offset = scroll_offset.saturating_add(usize::from(hidden_left));
+    visible.extend(token.chars().skip(content_offset).take(visible_width));
+    if hidden_right {
+        visible.push('…');
+    }
+    visible
```
[the remark this answers is not in the document]
Comment (Bortlesboat) [reply crates/mesh-llm-tui/src/output/mod.rs:4756]: The module is already about 16,000 lines on `main`; extracting this roughly 50-line renderer would not bring it under the 2,000-line guideline. A compliant split would require a broad refactor of unrelated dashboard code, so I have kept this issue-linked fix focused. I did add the requested empty-token and zero-width branch coverage in d4c4409d.

**Review by i386 — APPROVED** [review]
Reviewed the change and the edge cases; no changes needed.
